From bcba7281bbb2351fab69498c54fcda4e6154fa91 Mon Sep 17 00:00:00 2001
From: Laurent Vivier <lvivier@redhat.com>
Date: Fri, 23 Aug 2019 09:13:02 +0100
Subject: [PATCH 4/5] redhat: update pseries-rhel8.1.0 machine type

RH-Author: Laurent Vivier <lvivier@redhat.com>
Message-id: <20190823091302.8970-1-lvivier@redhat.com>
Patchwork-id: 90137
O-Subject: [RHEL-AV-8.1.0 qemu-kvm PATCH] redhat: update pseries-rhel8.1.0 machine type
Bugzilla: 1744170
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1744170
BRANCH: rhel-av-8.1.0/master-4.1.0
UPSTREAM: downstream only
BREW: https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=23160686

pseries-rhel8.1.0 has been created based on pseries-4.0,
we need to update it now we are based on pseries-4.1

Signed-off-by: Laurent Vivier <lvivier@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/ppc/spapr.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index 669eae1..9b500d2 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -4437,6 +4437,7 @@ static void spapr_machine_4_1_class_options(MachineClass *mc)
 }
 
 DEFINE_SPAPR_MACHINE(4_1, "4.1", true);
+#endif
 
 /*
  * pseries-4.0
@@ -4453,6 +4454,7 @@ static void phb_placement_4_0(SpaprMachineState *spapr, uint32_t index,
     *nv2atsd = 0;
 }
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 static void spapr_machine_4_0_class_options(MachineClass *mc)
 {
     SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
@@ -4796,7 +4798,7 @@ DEFINE_SPAPR_MACHINE(rhel810, "rhel8.1.0", true);
 
 /*
  * pseries-rhel8.0.0
- * like spapr_compat_3_1
+ * like pseries-3.1 and pseries-4.0
  * except SPAPR_CAP_CFPC, SPAPR_CAP_SBBC and SPAPR_CAP_IBS
  * that have been backported to pseries-rhel8.0.0
  */
@@ -4809,6 +4811,12 @@ static void spapr_machine_rhel800_class_options(MachineClass *mc)
     compat_props_add(mc->compat_props, hw_compat_rhel_8_0,
                      hw_compat_rhel_8_0_len);
 
+    /* pseries-4.0 */
+    smc->phb_placement = phb_placement_4_0;
+    smc->irq = &spapr_irq_xics;
+    smc->pre_4_1_migration = true;
+
+    /* pseries-3.1 */
     mc->default_cpu_type = POWERPC_CPU_TYPE_NAME("power8_v2.0");
     smc->update_dt_enabled = false;
     smc->dr_phb_enabled = false;
-- 
1.8.3.1

